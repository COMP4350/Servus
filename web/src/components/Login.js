import React from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import { Button, TextField } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    margin: '10%'
  },
  textField : {
    width: '50%',
    marginBottom: '3%'
  },
  button : {
    marginBottom: '3%'
  }
}));
const Login = () => {
  const classes = useStyles();
  const history = useHistory();
  const validateLogin = () => {
    history.push('/home');
    // validate login
  };

  return (
    
    <div className={classes.root}>
      <h2>Login</h2>
      <TextField className={classes.textField}
      id="outlined-basic"
      label="username"
      />
    <TextField className={classes.textField}
      id="outlined-password-input"
      label="password"
      type="password"
      autoComplete="current-password"
    />
    <Button className={classes.button}
      variant="contained"
      onClick={validateLogin}>
      LOGIN
    </Button>
    <Button className={classes.button}
      variant="contained"
      onClick={() => history.push('/signup')}>
      SIGN UP
  </Button>
  </div>)
};

export default Login;