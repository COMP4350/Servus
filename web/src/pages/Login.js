import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles, createMuiTheme } from '@material-ui/core/styles';
import {
    Button,
    TextField,
    ThemeProvider,
    Typography,
} from '@material-ui/core';
import axios from 'axios';
import useForm from '../hooks/useForm';

const theme = createMuiTheme({
    palette: {
        type: 'dark',
        primary: {
            main: '#647AA3',
        },
        secondary: {
            main: '#EC5732',
        },
        default: {
            main: '#FFF2EB',
        },
    },
    background: {
        dark: '#151515',
        main: '#272727',
    },
    typography: {
        fontFamily: ['Roboto'],
    },
});

const useStyles = makeStyles(() => ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-evenly',
        alignItems: 'center',
        margin: '2%',
    },
    textField: {
        width: '50%',
        marginBottom: '3%',
    },
    button: {
        marginBottom: '3%',
        textTransform: 'capitalize',
    },
    title: {
        color: 'white',
    },
}));

const Login = props => {
    const classes = useStyles();
    const history = useHistory();
    const [errors, setErrors] = useState({});
    const [formValid, setFormValid] = useState(false);
    const [loginForm, onFormChange] = useForm({
        username: '',
        password: '',
    });
    const validate = () => {
        let errors = {};
        if (!loginForm.username) {
            errors.username = 'username is required';
        }
        if (!loginForm.password) {
            errors.password = 'password is required';
        }
        setErrors(errors);
        setFormValid(Object.getOwnPropertyNames(errors).length == 0);
    };
    const login = async () => {
        if (formValid) {
            axios
                .post(
                    `/user/${loginForm.username}/login`,
                    {
                        password: loginForm.password,
                    },
                    {
                        withCredentials: true,
                    }
                )
                .then(res => {
                    console.log(res.data.result.username);
                    props.setUsername(res.data.result.username);
                    history.push('/');
                })
                .catch(() => {
                    alert('User not found.');
                });
            return;
        }
    };

    useEffect(() => {
        login();
    }, [formValid]);

    return (
        <ThemeProvider theme={theme}>
            <div className={classes.root}>
                <Typography variant="h2" className={classes.title}>
                    Login
                </Typography>
                <TextField
                    className={classes.textField}
                    id="outlined-basic"
                    label="username"
                    name="username"
                    value={loginForm.username}
                    onChange={onFormChange}
                    error={errors.username}
                />
                <TextField
                    className={classes.textField}
                    id="outlined-password-input"
                    label="password"
                    type="password"
                    name="password"
                    autoComplete="current-password"
                    value={loginForm.password}
                    onChange={onFormChange}
                    error={errors.password}
                />
                <Button
                    className={classes.button}
                    variant="contained"
                    onClick={validate}>
                    Login
                </Button>
                <Button
                    className={classes.button}
                    variant="contained"
                    onClick={() => history.push('/signup')}>
                    Sign up
                </Button>
            </div>
        </ThemeProvider>
    );
};

export default Login;
