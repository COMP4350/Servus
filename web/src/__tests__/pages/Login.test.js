import React from 'react';
import { act, render } from '@testing-library/react';
import Login from '../../pages/Login';

describe('Login', () => {
    it('renders correctly', async () => {
        const promise = Promise.resolve();

        const { asFragment } = render(<Login />);

        expect(asFragment()).toMatchSnapshot();

        await act(() => promise);
    });
});
