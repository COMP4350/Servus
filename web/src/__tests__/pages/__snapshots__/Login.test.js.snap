// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <h2
      class="MuiTypography-root makeStyles-title-4 MuiTypography-h2"
    >
      Login
    </h2>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        username
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="outlined-basic"
          name="username"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="outlined-password-input"
        id="outlined-password-input-label"
      >
        password
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          autocomplete="current-password"
          class="MuiInputBase-input MuiInput-input"
          id="outlined-password-input"
          name="password"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Login
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Sign up
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
